### PR TITLE
Implement support for KV1 binary serialization with string tables

### PR DESCRIPTION
**Pull request by yaakov-h**
This is a bit of a different approach to #99.

### CODE REVIEW

File: ValveKeyValue/ValveKeyValue/StringTable.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            }
+
+            return reverse[value];
+        }
+
+        public void PrepareForSerialization()
+        {
+            if (reverse is not null)
+            {
+                return;
```
Comment (xPaw) [inline ValveKeyValue/ValveKeyValue/StringTable.cs:45]: Is this gonna cause issues if you serialize, change the kv and try to serialize again?

Actually not seeing a way for stringtable to be populated by the seializer.

File: ValveKeyValue/ValveKeyValue/StringTable.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                if (index >= lookup.Length)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index), index, "Index must be less than the number of strings in the table.");
+                }
+
+                return lookup.Span[index];
+            }
+        }
+
+        public int IndexOf(string value)
```
Comment (xPaw) [inline ValveKeyValue/ValveKeyValue/StringTable.cs:31]: Indexof usually implies O(n), should this be this[string]?

Comment (yaakov-h) [reply ValveKeyValue/ValveKeyValue/StringTable.cs:31]: Never heard of that before, but yes.

File: ValveKeyValue/ValveKeyValue/StringTable.cs (hunk 1 above)
Comment (yaakov-h) [reply ValveKeyValue/ValveKeyValue/StringTable.cs:45]: oh that's an interesting idea, should we dynamically populate a string table?

Comment (xPaw) [reply ValveKeyValue/ValveKeyValue/StringTable.cs:48]: If we want to support serialization, we kind of have to.

Comment (xPaw) [reply ValveKeyValue/ValveKeyValue/StringTable.cs:48]: We also need a public accessor to the list if that happens, because you need to store it somehow

**Review by xPaw — APPROVED** [review]
Seems fine.

**Comment by xPaw** [comment]
Apisurface does not seem to dump constructors?

**Review by xPaw — APPROVED** [review]
Lets release this, can always improve further later.

**Pull request merged**